### PR TITLE
bug: Add logic to handle and send 404 for unregistered routes and send pro…

### PR DESCRIPTION
**Pull request by chiraglulla**
…per error codes from server to client

### CODE REVIEW

File: controllers/video.go [hunk 1]
```diff
@@ -145,7 +150,9 @@ func GetVideos(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	`)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println("Error running select query for all video records.")
+		http.Error(w, "Error Retreiving Records", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:154]:
```suggestion
		http.Error(w, "Error retreiving videos", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 2]
```diff
@@ -126,6 +129,8 @@ func UploadVideo(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 			log.Println("Successfully uploaded chunks of", fileName, "to Appwrite Storage")
 		} else {
 			log.Println("Error breaking " + fileName + " into .ts files.")
+			http.Error(w, "Error Processing File", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:132]:
```suggestion
			http.Error(w, "Error processing file", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 3]
```diff
@@ -162,7 +169,9 @@ func GetVideos(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 		err := rows.Scan(&id, &title, &description)
 
 		if err != nil {
-			log.Fatal(err)
+			log.Println("Error scanning rows")
+			http.Error(w, "Error Retreiving Records", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:173]:
```suggestion
			http.Error(w, "Error retreiving videos", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 4]
```diff
@@ -177,9 +186,12 @@ func GetVideos(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	recordsJSON, err := json.Marshal(records)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Records", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:190]:
```suggestion
		http.Error(w, "Error retreiving videos", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 5]
```diff
@@ -194,14 +206,20 @@ func GetVideo(w http.ResponseWriter, r *http.Request, db *sql.DB) {
...
+
 	defer detailsQuery.Close()
+
 	var title, description string
 	err = detailsQuery.QueryRow(video_id).Scan(&title, &description)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Record", http.StatusInternalServerError)
+		return
```
Comment (dheerajdlalwani) [inline controllers/video.go:222]: What happens when no rows are returned?
In case of no rows, return a 404 not found.

File: controllers/video.go [hunk 6]
```diff
@@ -214,10 +232,13 @@ func GetVideo(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	videoDetailsJSON, err := json.Marshal(videoDetails)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Record", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:236]:
```suggestion
		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 7]
```diff
@@ -230,7 +251,9 @@ func GetManifestFile(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	request, err := http.NewRequest("GET", getManifestFile, nil)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Video", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:255]:
```suggestion
		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
```

Comment (dheerajdlalwani) [reply controllers/video.go:255]: Also, if requested manifest file does not exist, return a 404

File: controllers/video.go [hunk 8]
```diff
@@ -243,18 +266,23 @@ func GetManifestFile(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	response, err := client.Do(request)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Video", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:270]:
```suggestion
		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
```

File: controllers/video.go [hunk 9]
```diff
@@ -243,18 +266,23 @@ func GetManifestFile(w http.ResponseWriter, r *http.Request, db *sql.DB) {
...
 	}
 
 	defer response.Body.Close()
 
 	bodyBytes, err := io.ReadAll(response.Body)
 
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		http.Error(w, "Error Retreiving Video", http.StatusInternalServerError)
```
Comment (dheerajdlalwani) [inline controllers/video.go:280]:
```suggestion
		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
```

**Review by dheerajdlalwani — CHANGES_REQUESTED** [review]
We have some fixes to look at.

**Comment by dheerajdlalwani** [comment]
What's the progress here, @chiraglulla ?

**Review by dheerajdlalwani — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (chiraglulla) [reply .vscode/settings.json:1]: It was already resolved in other branch and merged to main. As you can't see it in main now.

File: client/upload.html [hunk 10]
```diff
@@ -25,31 +25,49 @@
...
+          <label for="video">Drop that video...</label>
+          <input id="video" type="file" name="video" accept=".mp4" />
+          <br />
+  
+          <p id="fileError" class="special-red" style="display: none;"></p>
+  
+          <br />
+  
+          <!-- <label for="thumbnail">Thumbnail</label>
+          <input id="thumbnail" type="file" name="thumbnail" accept="image/*" /> -->
```
[the remark this answers is not in the document]
Comment (chiraglulla) [reply client/upload.html]: Done.

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (chiraglulla) [reply .vscode/settings.json:1]: Removed.

**Pull request closed without merging**